openedx/features/course_experience/course_updates.py

```diff
@@ -0,0 +1,112 @@
+"""
+Utilities for course updates.
+"""
+
+import hashlib
+from datetime import datetime
+
+from lms.djangoapps.courseware.courses import get_course_info_section_module
+from openedx.core.djangoapps.user_api.course_tag.api import get_course_tag, set_course_tag
+
+STATUS_VISIBLE = 'visible'
+STATUS_DELETED = 'deleted'
+
+VIEW_WELCOME_MESSAGE_KEY = 'view-welcome-message'
+
+
+def _calculate_update_hash(update):
+    """
+    Returns a hash of the content of a course update. Does not need to be secure.
+    """
+    hasher = hashlib.md5()
+    hasher.update(update['content'].encode('utf-8'))
+    return hasher.hexdigest()
+
+
+def _get_dismissed_hashes(user, course_key):
+    """
+    Returns a list of dismissed hashes, or None if all updates have been dismissed.
+    """
+    view_welcome_message = get_course_tag(user, course_key, VIEW_WELCOME_MESSAGE_KEY)
+    if view_welcome_message == 'False':  # legacy value, which dismisses all updates
+        return None
```

Comment:
By respecting previous values of `False`, this means that users that have dismissed an update will continue to never see new updates. But new enrollments will experience the new behavior of single-dismissals. Which I think is fine?